tests/browser/render/css.js

```diff
@@ -419,4 +419,171 @@ export default function() {
+
+		const style = getComputedStyle( fixture.querySelector( 'div' ) );
+
+		t.equal( style.color, 'rgb(0, 128, 0)' );
+		t.equal( style.backgroundColor, 'rgb(128, 0, 0)' );
+	});
+
+	test( `setting css for a component with applied styles updates the styles`, t => {
+		const cmp = Ractive.extend({
+			template: '<div />',
+			css () {
+				return `div { color: ${this.get('color') || 'rgb(255, 255, 255)'}; }`;
+			}
+		});
+
+		new cmp({ target: fixture });
+
+		const style = getComputedStyle( fixture.querySelector( 'div' ) );
+
+		t.equal( style.color, 'rgb(255, 255, 255)' );
+
+		cmp.setCSS( 'color', 'rgb(0, 128, 0)' );
+
+		t.equal( style.color, 'rgb(0, 128, 0)' );
+	});
+
+	test( `setting css for a component with applied styles can skip the update with an option`, t => {
+		const cmp = Ractive.extend({
+			template: '<div />',
+			css () {
+				return `div { color: ${this.get('color') || 'rgb(255, 255, 255)'}; }`;
+			}
+		});
+
+		new cmp({ target: fixture });
+
+		const style = getComputedStyle( fixture.querySelector( 'div' ) );
+
+		t.equal( style.color, 'rgb(255, 255, 255)' );
+
+		cmp.setCSS( 'color', 'rgb(0, 128, 0)', { apply: false } );
+
+		t.equal( style.color, 'rgb(255, 255, 255)' );
+		t.equal( cmp.cssData.color, 'rgb(0, 128, 0)' );
+	});
+
+	test( `setting css for a parent component cascades to children`, t => {
+		const cmp = Ractive.extend({
+			cssData: { color: 'rgb(0, 128, 0)' }
+		});
+		const cmp2 = cmp.extend({
+			template: '<div />',
+			css () {
+				return `div { color: ${ this.get( 'color' ) }; }`;
+			}
+		});
+
+		new cmp2({ target: fixture });
+
+		const style = getComputedStyle( fixture.querySelector( 'div' ) );
+
+		t.equal( style.color, 'rgb(0, 128, 0)' );
+
+		cmp.setCSS( 'color', 'rgb(255, 255, 255)' );
+
+		t.equal( style.color, 'rgb(255, 255, 255)' );
+	});
+
+	test( `css set cascade from parent can be stopped with an option`, t => {
+		const cmp = Ractive.extend({
+			cssData: { color: 'rgb(0, 128, 0)' }
+		});
+		const cmp2 = cmp.extend({
+			template: '<div />',
+			css () {
+				return `div { color: ${ this.get( 'color' ) }; }`;
+			}
+		});
+
+		new cmp2({ target: fixture });
+
+		const style = getComputedStyle( fixture.querySelector( 'div' ) );
+
+		t.equal( style.color, 'rgb(0, 128, 0)' );
+
+		cmp.setCSS( 'color', 'rgb(255, 255, 255)', { cascade: false } );
+
+		t.equal( style.color, 'rgb(0, 128, 0)' );
+	});
+
+	test( `all relevant component ids are applied to top-level nodes`, t => {
+		const cmp = Ractive.extend({
+			cssId: 'cmp',
+			css: `div { color: rgb(0, 128, 0); }`
+		});
+
+		const cmp2 = cmp.extend({
+			cssId: 'cmp2',
+			css: `div { display: inline-block; }`,
+			template: '<div />'
+		});
+
+		new cmp2({ target: fixture });
+
+		const div = fixture.querySelector( 'div' );
+		const style = getComputedStyle( div );
+
+		t.equal( div.getAttribute( 'data-ractive-css' ), '{cmp} {cmp2}' );
+		t.equal( style.color, 'rgb(0, 128, 0)' );
+		t.equal( style.display, 'inline-block' );
+	});
 }
```

Comment:
Might want to include tests for `toCSS` and `getCSS` for SSR purposes, and to pin down order-related things.